blog/you-cant-create-a-constant-using-usememo-in-react.md

```diff
@@ -0,0 +1,58 @@
+# You can't create a constant using useMemo in React
+
+There are 3 possible options to create a constant in React: `useState`, `useMemo` and `useRef`. We're going to see which one is the best way to create real constants (not constant-like values).
+
+I talked about the [`useConst`](https://github.com/chakra-ui/chakra-ui/blob/main/packages/hooks/src/use-const.ts) hook from [chakra-ui](https://chakra-ui.com) in a [previous post](https://dev.to/cloudx/chakra-ui-the-hidden-treasures-1gl9), and now, I found the same hook in [Fluent UI](https://www.npmjs.com/package/@fluentui/react-hooks#useconst), we are going to review why they've implemented it.
+
+## Remember useMemo
+
+The first thought that comes to mind when we look for a constant in a render function is to use the [`useMemo`](https://reactjs.org/docs/hooks-reference.html#usememo) hook. This is a good option, but only if your "constant" value shouldn't change based on the dependencies.
+
+```javascript
+const memoizedValue = useMemo(
+  () => computeExpensiveValue(a, b),
+  [a, b]
+);
+```
+
+There is no warranty that the returned value of `useMemo` is always the same even when the dependencies didn't change. React can run the factory function if the tool believes that the value should be re created.
```

Comment:
```suggestion
There is no guarantee that the returned value of `useMemo` is always the same even if the dependencies didn't change. React can run the factory function if the tool believes that the value should be re created.
```

Reply:
Could we add a connector between these sentences?

Reply:
```suggestion
There is no guarantee that the returned value of `useMemo` is always the same even when the dependencies didn't change. For this reason React runs the factory function if the tool believes that the value should be re created.
```